projects/igniteui-angular/src/lib/directives/mask/mask.directive.ts

```diff
@@ -105,264 +88,213 @@ export class IgxMaskDirective implements OnInit, ControlValueAccessor {
+        const format = this.mask || 'CCCCCCCCCC';
+        const promptChar = this.promptChar && this.promptChar.substring(0, 1);
+        return { format, promptChar };
     }
 
-    /**
-     *@hidden
-     */
-    private get selectionStart() {
-        return this.nativeElement.selectionStart;
+    private get selectionStart(): number {
+        // Edge(classic) and FF don't select text on drop
+        return this.nativeElement.selectionStart === this.nativeElement.selectionEnd && this._hasDropAction ?
+            this.nativeElement.selectionEnd - this._droppedData.length :
+            this.nativeElement.selectionStart;
     }
 
-    /**
-     *@hidden
-     */
-    private get selectionEnd() {
+    private get selectionEnd(): number {
         return this.nativeElement.selectionEnd;
     }
 
-    /**
-     *@hidden
-     */
-    private _ctrlDown: boolean;
-
-    /**
-     *@hidden
-     */
-    private _paste: boolean;
-
-    /**
-     *@hidden
-     */
-    private _selection: number;
-
-    /**
-     *@hidden
-     */
-    private _placeholder: string;
-
-    /**
-     *@hidden
-     */
-    private _maskOptions = {
-        format: '',
-        promptChar: ''
-    };
-
-    /**
-     *@hidden
-     */
-    private _key;
-
-    /**
-     *@hidden
-     */
-    private _cursorOnPaste;
-
-    /**
-     *@hidden
-     */
-    private _valOnPaste;
+    private get nativeElement(): HTMLInputElement {
+        return this.elementRef.nativeElement;
+    }
 
+    private _key: number;
+    private _oldVal: any;
+    private _end = 0;
+    private _start: number;
+    private _dataValue: string;
+    private _droppedData: string;
+    private _hasDropAction: boolean;
     private _stopPropagation: boolean;
 
-    /**
-     *@hidden
-     */
     private _onTouchedCallback: () => void = noop;
-
-    /**
-     *@hidden
-     */
     private _onChangeCallback: (_: any) => void = noop;
 
-    constructor(private elementRef: ElementRef, private maskParser: MaskParsingService) { }
+    constructor(
+        protected elementRef: ElementRef,
+        protected maskParser: MaskParsingService,
+        protected renderer: Renderer2) { }
 
-    /**
-     *@hidden
-     */
+    /** @hidden */
     public ngOnInit(): void {
-        if (this.promptChar && this.promptChar.length > 1) {
-            this._maskOptions.promptChar = this.promptChar = this.promptChar.substring(0, 1);
-        }
+        this.renderer.setAttribute(this.nativeElement, 'placeholder',
+            this.placeholder ? this.placeholder : this.maskOptions.format);
+    }
 
-        this._maskOptions.format = this.mask ? this.mask : 'CCCCCCCCCC';
-        this._maskOptions.promptChar = this.promptChar ? this.promptChar : '_';
-        this.nativeElement.setAttribute('placeholder', this.placeholder ? this.placeholder : this._maskOptions.format);
+    /** @hidden */
+    public ngAfterViewChecked(): void {
+        this._oldVal = this.inputValue;
```

Comment:
In theory definitely **not** the place to save old value. Kinda chatty hook this one. 
The mask is in control of this value, right? Why should it come from the native input, when you can save it when the mask assigns it there (afterInput, onFocus / showMask)?